[
  {
    "url": "https://api.github.com/repos/expressjs/express/issues/comments/239772809",
    "html_url": "https://github.com/expressjs/express/pull/3055#issuecomment-239772809",
    "issue_url": "https://api.github.com/repos/expressjs/express/issues/3055",
    "id": 239772809,
    "node_id": "MDEyOklzc3VlQ29tbWVudDIzOTc3MjgwOQ==",
    "user": {
      "login": "blakeembrey",
      "id": 1088987,
      "node_id": "MDQ6VXNlcjEwODg5ODc=",
      "avatar_url": "https://avatars.githubusercontent.com/u/1088987?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/blakeembrey",
      "html_url": "https://github.com/blakeembrey",
      "followers_url": "https://api.github.com/users/blakeembrey/followers",
      "following_url": "https://api.github.com/users/blakeembrey/following{/other_user}",
      "gists_url": "https://api.github.com/users/blakeembrey/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/blakeembrey/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/blakeembrey/subscriptions",
      "organizations_url": "https://api.github.com/users/blakeembrey/orgs",
      "repos_url": "https://api.github.com/users/blakeembrey/repos",
      "events_url": "https://api.github.com/users/blakeembrey/events{/privacy}",
      "received_events_url": "https://api.github.com/users/blakeembrey/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2016-08-15T10:28:15Z",
    "updated_at": "2016-08-15T10:30:36Z",
    "author_association": "MEMBER",
    "body": "@mykiimike I'm not sure why you couldn't catch it, but the error is forwarded on and always handled (either by the default Express error handler or your custom one).\n\nEdit: Also, you can't parse this using `url.parse`. It doesn't really make any sense to do that. You're trying to decode a random URL-encoded string, not parse a URL.\n",
    "reactions": {
      "url": "https://api.github.com/repos/expressjs/express/issues/comments/239772809/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/expressjs/express/issues/comments/239778629",
    "html_url": "https://github.com/expressjs/express/pull/3055#issuecomment-239778629",
    "issue_url": "https://api.github.com/repos/expressjs/express/issues/3055",
    "id": 239778629,
    "node_id": "MDEyOklzc3VlQ29tbWVudDIzOTc3ODYyOQ==",
    "user": {
      "login": "mykiimike",
      "id": 7666506,
      "node_id": "MDQ6VXNlcjc2NjY1MDY=",
      "avatar_url": "https://avatars.githubusercontent.com/u/7666506?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/mykiimike",
      "html_url": "https://github.com/mykiimike",
      "followers_url": "https://api.github.com/users/mykiimike/followers",
      "following_url": "https://api.github.com/users/mykiimike/following{/other_user}",
      "gists_url": "https://api.github.com/users/mykiimike/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/mykiimike/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/mykiimike/subscriptions",
      "organizations_url": "https://api.github.com/users/mykiimike/orgs",
      "repos_url": "https://api.github.com/users/mykiimike/repos",
      "events_url": "https://api.github.com/users/mykiimike/events{/privacy}",
      "received_events_url": "https://api.github.com/users/mykiimike/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2016-08-15T11:07:12Z",
    "updated_at": "2016-08-15T11:07:12Z",
    "author_association": "NONE",
    "body": "@blakeembrey Yes i just catch the real problem. So, yes url.parse() does not make sense. It fixs my problem and create new one. But this stills a problem. Just adding % in the request URI and the express process downs. \n\nThis is just because express tests must accept % without encoding (%25). It is a strange behavior of decodeURI()-like that does not catch % is an invalid unicode. BUT in the same way you will have problem if you put % followed of a valid unicode that the coder did not except :]. \n\n@blakeembrey i will look at the error handler\n",
    "reactions": {
      "url": "https://api.github.com/repos/expressjs/express/issues/comments/239778629/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/expressjs/express/issues/comments/239782699",
    "html_url": "https://github.com/expressjs/express/pull/3055#issuecomment-239782699",
    "issue_url": "https://api.github.com/repos/expressjs/express/issues/3055",
    "id": 239782699,
    "node_id": "MDEyOklzc3VlQ29tbWVudDIzOTc4MjY5OQ==",
    "user": {
      "login": "mykiimike",
      "id": 7666506,
      "node_id": "MDQ6VXNlcjc2NjY1MDY=",
      "avatar_url": "https://avatars.githubusercontent.com/u/7666506?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/mykiimike",
      "html_url": "https://github.com/mykiimike",
      "followers_url": "https://api.github.com/users/mykiimike/followers",
      "following_url": "https://api.github.com/users/mykiimike/following{/other_user}",
      "gists_url": "https://api.github.com/users/mykiimike/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/mykiimike/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/mykiimike/subscriptions",
      "organizations_url": "https://api.github.com/users/mykiimike/orgs",
      "repos_url": "https://api.github.com/users/mykiimike/repos",
      "events_url": "https://api.github.com/users/mykiimike/events{/privacy}",
      "received_events_url": "https://api.github.com/users/mykiimike/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2016-08-15T11:35:51Z",
    "updated_at": "2016-08-15T11:35:51Z",
    "author_association": "NONE",
    "body": "Thanks\n",
    "reactions": {
      "url": "https://api.github.com/repos/expressjs/express/issues/comments/239782699/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/expressjs/express/issues/comments/239958076",
    "html_url": "https://github.com/expressjs/express/pull/3055#issuecomment-239958076",
    "issue_url": "https://api.github.com/repos/expressjs/express/issues/3055",
    "id": 239958076,
    "node_id": "MDEyOklzc3VlQ29tbWVudDIzOTk1ODA3Ng==",
    "user": {
      "login": "blakeembrey",
      "id": 1088987,
      "node_id": "MDQ6VXNlcjEwODg5ODc=",
      "avatar_url": "https://avatars.githubusercontent.com/u/1088987?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/blakeembrey",
      "html_url": "https://github.com/blakeembrey",
      "followers_url": "https://api.github.com/users/blakeembrey/followers",
      "following_url": "https://api.github.com/users/blakeembrey/following{/other_user}",
      "gists_url": "https://api.github.com/users/blakeembrey/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/blakeembrey/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/blakeembrey/subscriptions",
      "organizations_url": "https://api.github.com/users/blakeembrey/orgs",
      "repos_url": "https://api.github.com/users/blakeembrey/repos",
      "events_url": "https://api.github.com/users/blakeembrey/events{/privacy}",
      "received_events_url": "https://api.github.com/users/blakeembrey/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2016-08-15T23:20:29Z",
    "updated_at": "2016-08-15T23:20:29Z",
    "author_association": "MEMBER",
    "body": "Can you create a reproduction of the issue crashing? I can't replicate the behaviour. I used:\n\n``` js\nvar express = require('express')\nvar app = express()\n\napp.get('/:param', function (req, res) {\n  res.send('success')\n})\n\napp.listen(3000)\n```\n\n![image](https://cloud.githubusercontent.com/assets/1088987/17682958/9c0fd4a6-6392-11e6-9a76-b92444a5a5be.png)\n",
    "reactions": {
      "url": "https://api.github.com/repos/expressjs/express/issues/comments/239958076/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  }
]